#165273432 User should be able to reset a password

joelmugaya opened the pull request:
#### What does this PR do?
This PR enables the user to reset his/her password.

#### Description of Task to be completed
- Create a password reset page
- Make a post to send an email address to the backend
- Post a new password to https://inviticus-staging.herokuapp.com/api/reset_password/<token>/
- Redirect the user to login page

#### How should this be manually tested?
- Navigate to `/password/reset` in the form, submit an email
- Navigate to `/resetpassword/<token>` enter new password and the token sent to your email and submit to update password.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#165273432 Host application on cloud](https://www.pivotaltracker.com/story/show/165273432)

#### Screenshots (if appropriate)
<img width="400" alt="Screenshot 2019-06-07 at 20 42 18" src="https://user-images.githubusercontent.com/49488163/59123071-0c258b00-8965-11e9-9bee-4172cbd6b733.png">

njeri-ngigi commented on src/app/ResetPassword/Components/PasswordReset.jsx:
```diff
@@ -0,0 +1,31 @@
+import Errors from '../../Utils/Containers/Errors';
+
+const PasswordReset = (props) => {
+  props = props.props;
```
use destructuring

njeri-ngigi commented on src/app/ResetPassword/Components/RequestPasswordReset.jsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
this file looks very similar to PasswordReset.jsx

njeri-ngigi commented on src/app/ResetPassword/Reducer.js:
```diff
@@ -0,0 +1,22 @@
+
+export default function (state = initialState, { type, payload }) {
+  switch (type) {
+  case resetpassword.RESET_PASSWORD_SUCCESS:
```
use destructuirng

njeri-ngigi commented on src/app/ResetPassword/action.js:
```diff
@@ -0,0 +1,52 @@
+import { resetpassword } from '../types';
+
+export const resetPasswordRequestAction = body => async (dispatch) => {
+  fetch('http://localhost:8000/api/reset_password/', {
```
consider using axios

joelmugaya committed:
feat(Passwordreset) User should be able to reset a password

- Create a password reset page
- Make a post to send an email address to the backend
- Post a new password to https://inviticus-staging.herokuapp.com/api/reset_password/<token>/
- Redirect the user to login page

[Starts #165273432]

njeri-ngigi commented on src/app/components/RequestPasswordReset.jsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
This file looks very similar to PasswordReset.jsx

njeri-ngigi commented on src/redux/actions/types.js:
```diff
@@ -0,0 +1,9 @@
+  RESET_PASSWORD_SUCCESS: 'RESET_PASSWORD_SUCCESS',
+  RESET_PASSWORD_FAILURE: 'RESET_PASSWORD_FAILURE'
+};
+export const URLs = {
```
These URLs will be reused a lot throughout the code. Place them in a file that's accessible

engjames commented on .env:
```diff
@@ -0,0 +1 @@
+REACT_APP_BACK_END_URL=https://inviticus-staging.herokuapp.com
```
Thanks for creating a separate for our backend URL but I think if it's better to create a new file inside our actions folder and import it where necessary. This is because defining it in .env will prevent others from using it since that file is ignored while commiting. Thanks

engjames commented on app.json:
```diff
@@ -0,0 +1,17 @@
+      }
+    ],
+    "stack": "heroku-18"
+  }
```
A new line is need here, please

engjames commented on src/redux/actions/types.js:
```diff
@@ -0,0 +1,9 @@
+};
+export const URLs = {
+  BACK_END1: 'https://inviticus-staging.herokuapp.com/',
+  BACK_END: 'http://localhost:8000/',
```
Please remove this link for localhost if we not going to use it remotely. Thanks

KabohaJeanMark commented on src/App.scss:
```diff
@@ -7,3 +7,37 @@
+  text-align: left;
+  color: #721c24;
+  background-color: #f8d7da;
+  border-color: #f5c6cb;
```
He fixed these

KabohaJeanMark reviewed (approved):
Great work Joel. I have checked this out and it works quite well. Great job on increasing the coverage.

njeri-ngigi reviewed (approved):
Good job on the tests, coverage is good.
Your code is also clean 👏

njeri-ngigi commented:
Awesome job @joelmugaya
